### PR TITLE
Todo List - List Interaction

### PR DESCRIPTION
**Pull request by davidolanrewaju**
# 📄 Todo List

>This PR was created to detail the work done in this branch. In this branch, interactions were added to remove all checked tasks and update the list.

## 🛠Built With:
- HTML
- CSS
- JS
- Github Flows
- Linters (webhint, eslint, stylelint, lighthouse)
- Webpack (module bundler)

## ♻Implemented Features:
- [x] Add tasks dynamically by hardcoding
- [x] Check if the checkboxes have been clicked or not
- [x] CRUD functions
- [x] Remove all checked tasks and update the list

### CODE REVIEW

File: src/modules/todolist.js [hunk 1]
```diff
@@ -1,9 +1,11 @@
 /* eslint-disable import/no-cycle */
```
Comment (amon-cofie) [inline src/modules/todolist.js:1]: You have done very well on this project 👍 However, the linter checks are in place to ensure coding best practices and also ensure that you don't encounter bugs or security issues when you scale up your project. I suggest you do not disable them but fix the issues that are pointed out. I can assure you that this is a fix that is worth it and very important. 💯

File: src/modules/todolist.js [hunk 2]
```diff
@@ -34,11 +36,7 @@ function renderList() {
 
     const checkbox = document.querySelector(`input[data-index="${i}"]`);
     // eslint-disable-next-line no-loop-func
```
Comment (amon-cofie) [inline src/modules/todolist.js:38]: Please refer to my first comment 👍

File: src/modules/todolist.js [hunk 3]
```diff
@@ -79,6 +77,7 @@ refreshBtn.addEventListener('click', () => {
   location.reload();
   loadData();
 });
+clearBtn.addEventListener('click', clearCompleted);
 
 // eslint-disable-next-line import/prefer-default-export
```
Comment (amon-cofie) [inline src/modules/todolist.js:82]: Please refer to my first comment 👍

File: src/modules/update-list.js [hunk 4]
```diff
@@ -0,0 +1,25 @@
+/* eslint-disable-next-line import/no-cycle */
```
Comment (amon-cofie) [inline src/modules/update-list.js:1]: Please refer to my first comment 👍

File: src/modules/update-list.js [hunk 5]
```diff
@@ -0,0 +1,25 @@
...
+  // update the index property of the remaining tasks
+  todoList.forEach((task, i) => {
+    task.index = i + 1;
+  });
+
+  saveData();
+  renderList();
+}
+
+// eslint-disable-next-line import/prefer-default-export
```
Comment (amon-cofie) [inline src/modules/update-list.js:24]: Please refer to my first comment 👍

**Review by amon-cofie — CHANGES_REQUESTED** [review]
Hi @dave-prog , 👋 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there! 💪 



## Highlights

🚀 You have a descriptive pull request summary
🚀 You have good commit messages
🚀 You used GitHub flow properly
🚀 You have no linter errors

## Required Changes ♻️

- [ ] Everything works as is supposed to and your code is well structure 👍 Very good work done 💯. However, you have linter checks disabled in your code and that can be detrimental when you have to scale up. Please don't disable the linter checks. 

_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
- x
- y
- z

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/modules/update-list.js (hunk 5 above)
Comment (davidolanrewaju) [reply src/modules/update-list.js:24]: ESLint was disabled for those parts of the code because terms the import as no named modules

**Review by franclobo — APPROVED** [review]
Hi @dave-prog, 

# STATUS: APPROVED 😃 

git-merge

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights

- No linter errors ✔️ 
- Descriptive PR ✔️ 
- Correct Github flow ✔️ 
- Array.filter() method is used ✔️ 

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**